CONTRIBUTING.md

```diff
@@ -0,0 +1,513 @@
+
+A non-exhaustive list of examples of consensus-critical changes include:
+
+* Adding or changing block, microblock, or transaction wire formats
+* Changing the criteria under which a burnchain operation will be accepted by the node
+* Changing the data that gets stored to a MARF key/value pair in the Clarity or Stacks chainstate MARFs
+* Changing the order in which data gets stored in the above
+* Adding, changing, or removing Clarity functions
+* Changing the cost of a Clarity function
+* Adding new kinds of transactions, or enabling certain transaction data field values that were previously forbidden.
+
+## Testing
+
+* **Unit tests should focus on the business logic with mocked data**.  To the greatest extent possible, each error path should be tested _in addition to_ the success path.  A submitter should expect to spend most of their test-writing time focusing on error paths; getting the success path to work is often much easier than the error paths.
+
+* **Unit tests should verify that the I/O code paths work**, but do so in a way that does not "clobber" other tests or prevent other tests from running in parallel (if it can be avoided).  This means that unit tests should use their own directories for storing transient state (in `/tmp`), and should bind on ports that are not used anywhere else.
+
+* If randomness is needed, **tests should use a seeded random number generator if possible**.  This ensures that they will reliably pass in CI.
+
+* When testing a consensus-critical code path, the test coverage should verify that the new behavior is only possible within the epoch(s) in which the behavior is slated to activate.  Above all else, **backwards-compatibility is a hard requirement.**
+
+* **Integration tests are necessary when the PR has a consumer-visible effect**.  For example, changes to the RESTful API, event stream, and mining behavior all require integration tests.
+
+* Every consensus-critical change needs an integration test to verify that the feature activates only when the hard fork activates.
+
+PRs must include test coverage. However, if your PR includes large tests or tests which cannot run in parallel
+(which is the default operation of the `cargo test` command), these tests should be decorated with `#[ignore]`.
+
+A test should be marked `#[ignore]` if:
+
+  1. It does not _always_ pass `cargo test` in a vanilla environment
+     (i.e., it does not need to run with `--test-threads 1`).
+
+  2. Or, it runs for over a minute via a normal `cargo test` execution
+     (the `cargo test` command will warn if this is not the case).
+
+
+
+## Formatting
+
+This repository uses the default rustfmt formatting style. PRs will be checked against `rustfmt` and will _fail_ if not
+properly formatted.
+
+You can check the formatting locally via:
+
+```bash
+cargo fmt --all -- --check
+```
+
+You can automatically reformat your commit via:
+
+```bash
+cargo fmt --all
```

Comment:
should be `cargo fmt --all -- --config group_imports=StdExternalCrate` and `cargo fmt --all -- --check --config group_imports=StdExternalCrate`

Reply:
Fixed in 2b5cda5b6